kernels/cpu/include/readdy/kernel/cpu/actions/reactions/ReactionUtils.h

```diff
@@ -70,7 +73,8 @@ inline bool shouldPerformEvent(const double rate, const double timestep, bool ap
 data_t::update_t handleEventsGillespie(
         CPUKernel *const kernel, double timeStep,
         bool filterEventsInAdvance, bool approximateRate,
-        std::vector<event_t> &&events, std::vector<record_t>* maybeRecords, reaction_counts_t* maybeCounts);
+        std::vector<event_t> &&events, std::vector<record_t> *maybeRecords, reaction_counts_order1_map *maybeCountsOrder1,
```

Comment:
i would vouch for tuple here, as you'd either want to record both orders or none